src/search/indexing/local_import.rs

```diff
@@ -1,70 +1,108 @@
-        .find(None, None)
-        .await
-        .map_err(DatabaseError::LocalDatabaseError)?;
+    let mut results = sqlx::query!(
+        "
+        SELECT m.id, m.title, m.description, m.downloads, m.icon_url, m.body_url, m.published FROM mods m
+        "
+    )
+    .fetch(&pool);
 
-    while let Some(unparsed_result) = results.next().await {
-        let result: Mod =
-            *Mod::from_doc(unparsed_result.map_err(DatabaseError::LocalDatabaseError)?)?;
+    while let Some(result) = results.next().await {
+        if let Ok(result) = result {
+            /*
+            let result: Mod =
+                *Mod::from_doc(unparsed_result.map_err(DatabaseError::LocalDatabaseError)?)?;
 
-        let mut mod_versions = versions
-            .find(doc! { "mod_id": result.id }, None)
-            .await
-            .map_err(DatabaseError::LocalDatabaseError)?;
+            let mut mod_versions = versions
+                .find(doc! { "mod_id": result.id }, None)
+                .await
+                .map_err(DatabaseError::LocalDatabaseError)?;
 
-        let mut mod_game_versions = vec![];
+            let mut mod_game_versions = vec![];
 
-        while let Some(unparsed_version) = mod_versions.next().await {
-            let mut version = unparsed_version
-                .map_err(DatabaseError::LocalDatabaseError)
-                .and_then(Version::from_doc)?;
-            mod_game_versions.append(&mut version.game_versions);
-        }
+            while let Some(unparsed_version) = mod_versions.next().await {
+                let mut version = unparsed_version
+                    .map_err(DatabaseError::LocalDatabaseError)
+                    .and_then(Version::from_doc)?;
+                mod_game_versions.append(&mut version.game_versions);
+            }
+            */
 
-        let mut icon_url = "".to_string();
+            let versions: Vec<String> = sqlx::query!(
+                "
+                SELECT gv.version FROM versions
+                    INNER JOIN game_versions_versions gvv ON gvv.joining_version_id=versions.id
+                    INNER JOIN game_versions gv ON gvv.game_version_id=gv.id
+                WHERE versions.mod_id = $1
+                ",
+                result.id
+            )
+            .fetch_all(&pool)
+            .await?
+            .into_iter()
+            .map(|c| c.version)
+            .collect::<Vec<String>>();
 
-        if let Some(url) = result.icon_url {
-            icon_url = url;
-        }
+            let categories = sqlx::query!(
+                "
+                SELECT c.category
+                FROM mods_categories mc
+                    INNER JOIN categories c ON mc.joining_category_id=c.id
+                WHERE mc.joining_mod_id = $1
+                ",
+                result.id
+            )
+            .fetch_all(&pool)
+            .await?
+            .into_iter()
+            .map(|c| c.category)
+            .collect::<Vec<String>>();
```

Comment:
Is there a better way of doing this that doesn't require allocating two `Vec`s?  It seems like it should be possible using streams and stream combinators.